### ISSUE TITLE
Request - Functional Admins

### ISSUE THREAD

**Issue by johnboyjjb** [MegaMek/mekhq#2360]
Admins in MekHQ are required to not tank the unit score and are handy when negotiating contracts but tend to have limited value elsewhere.

This request is to consider having the logistics admin assist with the ordering of spare parts. Either notifying when inventory goes below a set threshold or auto order when below a threshold.

Likewise, the HR admin could notify when an employee has crossed a threshold for promotion and/or when an employee has enough XP to improve a skill (either the predefined next skill, or a defined skill tree, or perhaps just any skill the employee already has)

**Comment by purpledragon2** [comment]
Agreed.  I have house rules for the allowance of extra hires/purchases/tries for better maintenance sites.  

These are the things I am doing:
The admin/transport officer acquires and keeps the site(s) for the units.  
The admin/Logistics officer allows for purchases beyond what's in the unit market.
The admin/command officer allows for looking at potential hires outside of the personnel market.
Admin/HR personnel with Leadership may be placed offsite on other planets to run "Company owned" assets/businesses to try and generate extra income for the unit.  

If anyone is interested in trying to make heads or tails out of my house rule notes concerning this, send me a message and I will make it available.

**Comment by mjbroekman** [comment]
Admins are exceedingly useful if you train them in the appropriate acquisition skill and have a limit to the number of acquisitions per person per day.

While there is no distinction between that roles _do_, I've been balancing my admins across the different roles and, currently, have 14 Administrators and 10 Mechwarriors (my elite pilots that need extra ways to gain XP) trained in 'Administration' and 'Negotiation'. This results in procurement happening really fast since there are 24 different people able to perform up to 4 acquisition checks per day.

As an alternative to the house rules outlined by @purpledragon2 , maybe the roles could be broken out as follows:
- Admin/Command: used for rerolling scenario parameters (maybe influenced by their Strategy skill)
- Admin/HR : used for hiring recruits, purchasing infantry/BA (maybe with prices influenced by their Leadership and Negotiation skills)
- Admin/Logistical : used for acquisition checks for parts and refit kits (maybe with prices influenced by their Scrounge and Negotiation skills)
- Admin/Transport : used for purchasing vehicles, meks, etc (with prices influenced by their Negotiation skill)

Additionally, during mission/contract negotiations:
- Admin/Command : used to find contracts and negotiate everything not listed below
- Admin/HR : used to reroll command structure terms
- Admin/Logistical : used to reroll support costs
- Admin/Transport : used to reroll transportation terms

**Comment by johnboyjjb** [comment]
The integration point that I was thinking of into MekHQ was to have a the option to tick a box next to certain items and mark them as "Always have in stock". Then if you have enough logistics admins, they will order parts automatically to bring inventory up to the predefined level. Frequently I am spending a lot of time not blowing things up because I don't have some staple of the unit like armor or ammo. It's times like these I wonder why I am paying all these logistics admins.

Likewise, with a few hundred employees in my unit, managing promotions and skill upgrades are pretty slick and easy, but so time consuming. And I have to wonder why I keep paying these HR admins. I like the idea of a table that shows a route to my "perfect" mechwarrior that the admins will throw a message into the log that says, "Hey, mechwarrior Billy Bob has enough XP to increase his Gunnery skill" "Hey, admin Jim Bob has enough XP to increase his Negotiation skill"

**Comment by mjbroekman** [comment]
Along those lines, it would be great to be able to set Rank / Skill Level associations to auto-promote people when they attain the appropriate skill level.

Example:
  Admin/Command: Green => Private, Regular => Corporal, Veteran => Lieutenant JG, Elite => Captain
  Mechwarrior => Green => Recruit, Regular => Private, Veteran => Sergeant, Elite => Lieutenant SG
  Tech => Green => None, Regular => Corporal, Veteran+ => Master Sergeant

Admin Jim Bob has become a 'Veteran' Admin. "Great! Congrats, Jim Bob on your promotion to Lieutenant JG!".
Mechwarrior Joe Bob has become a 'Regular' Mechwarrior. "Great! Congrats, Joe Bob on your promotion to Private!". 
MechTech Bill Bob has become an 'Elite' MechTech. "Great! Congrats, Bill Bob. No promotion for you."

**Comment by rroby1** [comment]
I've been doing some of this "outside" of HQ using an Excel spreadsheet. Does help with the identifying of promotions / training, etc, but still requires a bit of manual intervention in HQ to make the changes. Back when the pandemic started I decided I wanted to learn a new skill while I wasn't able to go out and do what I would normally be doing. I started to learn Python. Been using HQ to help. Coded a Python script that takes a BLK file and determines the price of a "new" mech. I've recently been putting together thoughts about how to automate my administrative functions currently in the Excel file and manual changes in HQ.

**Comment by HammerGS** [comment]
@IllianiCBT Just tagging you as somethings here might cross over with planned work/ideas

**Comment by IllianiCBT** [comment]
I like the idea of Admin/Logi working to maintain stock minimums, but would take a *lot* of work to implement. I’d like to incorporate that into a ‘click button, order minimums’ system because darn if that would be useful. Not committing to doing this, but it is something I’m interested in seeing happen.

I think, in general, there does need to be some work done on how personnel procure new items. As it currently stands, this is a disproportionate source of XP compared to other sources. I see a lot of users giving personnel the secondary role of Admin/Logi just to rack up that XP which seems unintended.

Regarding Admin/Command, and Admin/Transport, I would like to see personnel other than Best in Role influencing negotiation re-rolls. That’s something I’m experimenting with in the Turnover & Retention Module and would be a relatively easy change to make. Potentially using a system similar to Administrative Capacity where the more TOE entries you have the more Admin/Command needed; and the more cargo you have the more Admin/Transport needed. Something that awards and requires having more than one Admin/Command and Admin/Transport, and reduces the temptation for recruiting 101,000,000 Admin/Logi

**Comment by IllianiCBT** [comment]
All Admin specializations had functionality added in 2024 to give them distinct roles and purpose. Closing.
